### PR TITLE
feat: add a dynamic type check method for plot API

### PR DESCRIPTION
**Pull request by taro-yu**
## Description

This PR adds a dynamic type check method for the plot API. Previously, @type_check_decorator was used, but this PR was created because individual type checking is now required due to modifications to the plot API annotations.

In addition, two annotations, Publisher and Subscription, were added to the histogram API for period and frequency in the previous (https://tier4.atlassian.net/browse/RT2-1203). However, additional modifications were required to plot these, so they have also been made.

image

## Related links

https://tier4.atlassian.net/browse/RT2-1204

## Pre-review checklist for the PR author

- [x] I've confirmed the [contribution guidelines](https://github.com/tier4/caret/blob/main/.github/CONTRIBUTING.md).
- [x] The PR is ready for review.

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR has been properly tested.
- [x] The PR has been reviewed.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] (Optional) The PR has been properly tested with CARET_report verification.
- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

### CODE REVIEW

File: src/caret_analyze/common/type_check_decorator.py [hunk 1]
```diff
@@ -185,26 +189,65 @@ def _get_given_arg_type(
...
+
     def decorator(func):
         validate_arguments_wrapper = \
             validate_arguments(config={'arbitrary_types_allowed': True})(func)
 
         @wraps(func)
         def _custom_wrapper(*args, **kwargs):
             try:
+                # Checks whether the arguments of a given func have variable length arguments
+                check_exist_varargs = inspect.getfullargspec(func)
```
Comment (rokamu623) [inline src/caret_analyze/common/type_check_decorator.py:235]: This variable is not appropriate.

`check_xxx` looks like it is an boolean value.
If this variable has other type, it should be named such that another programmer can know what kind of data it has.

File: src/test/common/test_type_check_decorator.py [hunk 2]
```diff
@@ -113,3 +119,17 @@ def kwarg(k: bool):
...
             kwarg(k=10)
         assert "'k' must be 'bool'. The given argument type is 'int'" in str(e.value)
+
+    def test_type_check_decorator_variable_length_arg(self):
+        @type_check_decorator
+        def var_len_args(*i: DummyCustom1 | DummyCustom2):
+            pass
+
+        dummy_1 = DummyCustom1
+        dummy_2 = DummyCustom2
```
[the remark this answers is not in the document]
Comment (taro-yu) [reply src/test/common/test_type_check_decorator.py]: Fixed in https://github.com/tier4/caret_analyze/pull/472/commits/189c4949dc6b47fd9441d8198e464a7cc0c83b58.

File: src/caret_analyze/common/type_check_decorator.py (hunk 1 above)
Comment (taro-yu) [reply src/caret_analyze/common/type_check_decorator.py]: Fixed in https://github.com/tier4/caret_analyze/pull/472/commits/189c4949dc6b47fd9441d8198e464a7cc0c83b58.

**Review by rokamu623 — APPROVED** [review]
LGTM!

**Review by xygyo77 — APPROVED** [review]
image

LGTM

**Pull request merged**